docs/source/tutorial/test_a_project.md

```diff
@@ -0,0 +1,335 @@
+         "price": [120, 290, 30]})
+
+@pytest.fixture
+def dummy_parameters():
+    parameters = {"model_options":
+                     {"test_size": 0.2,
+                      "random_state": 3,
+                      "features": ["engines", "passenger_capacity", "crew"]}
+                 }
+    return parameters
+```
+
+</details>
+
+We can then access these through the test arguments.
+
+```python
+def test_split_data(self, dummy_data, dummy_parameters):
+        ...
+```
+
+### Pipeline slicing
+
+In the test `test_data_science_pipeline` we test the that outputs of the data science pipeline are those that are expected. However, as pipelines are not static, this test is not robust. Instead we should be specific with how we define the pipeline to be tested; we do this by using [pipeline slicing](../nodes_and_pipelines/slice_a_pipeline.md#slice-a-pipeline-by-running-specified-nodes) to specify the pipeline's start and end:
+
+```python
+    def test_data_science_pipeline(self):
+        # Arrange pipeline
+        pipeline = create_pipeline().from_nodes("split_data_node").to_nodes("evaluate_model_node")
+        ...
+```
+
+This ensures that the test will still perform as designed, even with the addition of more nodes to the pipeline.
+
+
+After incorporating these testing practices, our test file `test_data_science.py` becomes:
+
+<details>
+
+```python
+# tests/pipelines/test_data_science_pipeline.py
+
+import pandas as pd
+import pytest
+
+from kedro.io import DataCatalog, MemoryDataset
+from kedro.runner import SequentialRunner
+from spaceflights.pipelines.data_science import create_pipeline
+from spaceflights.pipelines.data_science.nodes import split_data
+
+@pytest.fixture
+def dummy_data():
+    return pd.DataFrame(
+        {"engines": [1, 2, 3],
+         "crew": [4, 5, 6],
+         "passenger_capacity": [5, 6, 7],
+         "price": [120, 290, 30]})
+
+@pytest.fixture
+def dummy_parameters():
+    parameters = {"model_options":
+                     {"test_size": 0.2,
+                      "random_state": 3,
+                      "features": ["engines", "passenger_capacity", "crew"]}
+                 }
+    return parameters
+
+
+class TestDataScienceNodes:
+    def test_split_data(self, dummy_data, dummy_parameters):
+        X_train, X_test, y_train, y_test = split_data(dummy_data, dummy_parameters["model_options"])
+        assert len(X_train) == 2
+        assert len(y_train) == 2
+        assert len(X_test) == 1
+        assert len(y_test) == 1
+
+class TestDataSciencePipeline:
+    def test_data_science_pipeline(self, dummy_data, dummy_parameters):
+        pipeline = create_pipeline().from_nodes("split_data_node").to_nodes("evaluate_model_node")
+        catalog = DataCatalog()
+        catalog.add("model_input_table", MemoryDataset(dummy_data))
+        catalog.add_feed_dict({"params:model_options" : dummy_parameters["model_options"]})
+
+        output = SequentialRunner().run(pipeline, catalog)
```

Comment:
What is this output supposed to have? Maybe a bit more discussion on the runner in the previous paragraph would help

Reply:
The pipeline has no outputs, this was a trivial check to have an explicit assert that would confirm the pipeline ran successfully. I'll take a look at how to better capture that